examples/nlp/text_classification/text_classification_with_bert.py

```diff
@@ -12,203 +12,94 @@
-    parser.add_argument(
-        "--tokenizer",
-        default="nemobert",
-        type=str,
-        choices=["nemobert", "sentencepiece"],
-        help="tokenizer to use, only relevant when using custom pretrained checkpoint.",
-    )
-    parser.add_argument("--vocab_file", default=None, type=str, help="Path to the vocab file.")
-    parser.add_argument(
-        "--tokenizer_model",
-        default=None,
-        type=str,
-        help="Path to pretrained tokenizer model, only used if --tokenizer is sentencepiece",
-    )
-    parser.add_argument(
-        "--do_lower_case",
-        action='store_true',
-        help="Whether to lower case the input text. True for uncased models, False for cased models. "
-        + "For tokenizer only applicable when tokenizer is build with vocab file.",
-    )
-    parser.add_argument("--batch_size", default=32, type=int, help="Training and evaluation batch size")
-    parser.add_argument(
-        "--max_seq_length",
-        default=36,
-        type=int,
-        help="The maximum total input sequence length after tokenization.Sequences longer than this will be \
-                        truncated, sequences shorter will be padded.",
-    )
-
-    # Model Arguments
-    parser.add_argument("--num_output_layers", default=2, type=int, help="Number of layers in the Classifier")
-    parser.add_argument("--fc_dropout", default=0.1, type=float, help="Dropout rate")
-    parser.add_argument("--class_balancing", default="None", type=str, choices=["None", "weighted_loss"])
-
-    # Training Arguments
-    parser.add_argument("--gpus", default=1, type=int, help="Number of GPUs")
-    parser.add_argument("--num_nodes", default=1, type=int, help="Number of nodes")
-    parser.add_argument("--max_epochs", default=100, type=int, help="Total number of training epochs to perform.")
-    parser.add_argument(
-        "--max_steps",
-        default=None,
-        type=int,
-        help="Maximum number of steps to train. If it is set, num_epochs would get ignored.",
-    )
-    parser.add_argument(
-        "--accumulate_grad_batches", default=1, type=int, help="Accumulates grads every k batches.",
-    )
-    parser.add_argument(
-        "--amp_level", default="O0", type=str, choices=["O0", "O1", "O2"], help="01/02 to enable mixed precision"
-    )
-    parser.add_argument("--local_rank", default=None, type=int, help="For distributed training: local_rank")
-
-    # Validation Arguments
-    parser.add_argument("--save_epoch_freq", default=1, type=int, help="Epoch frequency of saving checkpoints")
-    parser.add_argument("--save_step_freq", default=-1, type=int, help="Step frequency of saving checkpoints")
-    parser.add_argument('--loss_step_freq', default=25, type=int, help='Frequency of printing loss')
-    parser.add_argument('--eval_epoch_freq', default=1, type=int, help='Frequency of evaluation')
-
-    args = parser.parse_args()
+@hydra.main(config_name="config")
+def main(cfg):
```

Comment:
Who passes this cfg to the main?

Reply:
The Hydra decorator.

Reply:
What is the alternative if we do not want to use this decorator?  It may look like magic to some users.
I think the name of the config file should get passed as an argument to the main so that the user can switch between configs?

Reply:
@VahidooX let's use this decorator for now. For "I think the name of the config file ..." I think there is solution such as config groups, etc. but let's worry about it after this PR

Reply:
@VahidooX There's no alternative to this decorator at the moment, and even it's replacement will just be another decorator. Hydra requires this.
As to switching config files, there's a way to do this which both Eric and I are looking into.